client/src/components/Connections/ListEspecificConnections.tsx

```diff
@@ -8,43 +8,40 @@ const ListEspecificConnections = () => {
   const params = useParams();
   const [connections, setConnections] = useState<any | any[]>([]);
 
-  const GetConnections = async () => {
-    try {
-      const response = await fetch("http://localhost:3000/connections" + '/' + params.id);
-      const jsonData = await response.json();
+  const GetConnections = () => {
 
-      const tuples: any[] = [];
-      await jsonData.map(async (connection:ConnectionInterface) => {
-        const u1 = await getUserNamePromise(connection.user1_id);
-        const u2 = await getUserNamePromise(connection.user2_id);
-        tuples.push({ user1_id: connection.user1_id, user1_name: u1, user2_name: u2, user2_id: connection.user2_id })
-        setConnections(tuples);
-      });
-    } catch (err) {
-      if(err instanceof Error){
-        console.error(err.message);
-    }else{
-        console.error("Unexpected error",err);
-    }
-    }
+    fetch("http://localhost:3000/connections" + '/' + params.id)
+      .then((res) => {
+        res.json()
+        .then((resJson) => {
+          const promisesArray = resJson.map((connection: ConnectionInterface) => {
+            const name1Promise = getUserNamePromise(connection.user1_id);
```

Comment:
I'm wondering why we make a call for each name instead of getting all of them in a single request. Will be good to review this behavior in the future.